core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -254,7 +254,6 @@ private static OAuthTokenResponse handleOAuthRequest(Object body) {
       case "client_credentials":
         return OAuthTokenResponse.builder()
             .withToken("client-credentials-token:sub=" + request.get("client_id"))
-            .withIssuedTokenType("urn:ietf:params:oauth:token-type:access_token")
```

Comment:
I would probably leave this as-is, since `issued_token_type` is required according to RFC 8693

Reply:
I don't agree: this flow is not governed by RFC 8693, this is RFC 6749 section 4.4.